bytelatent/metrics.py

```diff
@@ -54,14 +55,24 @@ class LoggingArgs(BaseModel):
 
 
 class MetricLogger:
-    def __init__(self, outdir: Path, args: Any | None = None):
+    def __init__(
+        self,
+        outdir: Path,
+        # args: TrainArgs
+        args: Any | None = None,
+        fs: fsspec.AbstractFileSystem | None = None,
+    ):
         self.outdir = outdir
         self.jsonl_writer = None
+        self.fs = fs
         self.args = args
 
     def open(self):
         if self.jsonl_writer is None:
-            self.jsonl_writer = open(self.outdir, "a")
+            if self.fs is None:
+                self.jsonl_writer = open(self.outdir, "a")
+            else:
+                self.jsonl_writer = self.fs.open(self.outdir, "a")
```

Comment:
What functionality does this add?

Reply:
Eventually (once we fix `torch.distributed.checkpointing.save` to work with `fsspec`), all of `dump_dir` should be compatible with writing to nfs or s3/blob store.